### ISSUE TITLE
[TSQL] Parser is unable to parse valid TSQL with FOR XML ('') ,TYPE

### ISSUE THREAD

**Issue by tkachenkomaria244** [sqlfluff/sqlfluff#5567]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The parser is unable to parse the following T-SQL query:
```SELECT (SELECT BusinessEntityID, FirstName, LastName, AdditionalContactInfo.query('
DECLARE namespace aci="https://schemas.microsoft.com/sqlserver/2004/07/adventure-works/ContactInfo";
DECLARE namespace act="https://schemas.microsoft.com/sqlserver/2004/07/adventure-works/ContactTypes";
//act:telephoneNumber/act:number
') AS PhoneNumbers
FROM Person.Person
FOR XML AUTO, TYPE).query('/Person.Person[1]');
```

### Expected Behaviour

Parsing is done fo this (and similar) constructions described in [basic-syntax-of-the-for-xml-clause](https://learn.microsoft.com/en-us/sql/relational-databases/xml/basic-syntax-of-the-for-xml-clause?view=sql-server-ver16)

### Observed Behaviour

```
...
[L:  7, P:  1]      |                                        for_clause:
[L:  7, P:  1]      |                                            keyword:          'FOR'
[L:  7, P:  4]      |                                            whitespace:       ' '
[L:  7, P:  5]      |                                            keyword:          'XML'
[L:  7, P:  8]      |                                            whitespace:       ' '
[L:  7, P:  9]      |                                            keyword:          'AUTO'
[L:  7, P: 13]      |                                unparsable:                   !! Expected: 'Nothing here.'
[L:  7, P: 13]      |                                    comma:                    ','
[L:  7, P: 14]      |                                    whitespace:               ' '
[L:  7, P: 15]      |                                    word:                     'TYPE'
[L:  7, P: 19]      |                                [META] dedent:
[L:  7, P: 19]      |                                end_bracket:                  ')'
[L:  7, P: 20]      |                    [META] dedent:
[L:  7, P: 20]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  7, P: 20]      |        dot:                                                  '.'
[L:  7, P: 21]      |        word:                                                 'query'
[L:  7, P: 26]      |        start_bracket:                                        '('
[L:  7, P: 27]      |        single_quote:                                         "'/Person.Person[1]'"
[L:  7, P: 46]      |        end_bracket:                                          ')'
[L:  7, P: 47]      |        semicolon:                                            ';'
[L:  7, P: 48]      |    [META] end_of_file:

==== parsing violations ====
L:   7 | P:  13 |  PRS | Line 7, Position 13: Found unparsable section: ', TYPE'
L:   7 | P:  20 |  PRS | Line 7, Position 20: Found unparsable section:
                       | ".query('/Person.Person[1]');"
```

### How to reproduce

Run the following command on the attached file:

`sqlfluff parse script_for_sqlfluff.sql --dialect tsql`

### Dialect

tsql

### Version

sqlfluff 2.3.5
Python 3.11.4

### Configuration

```[sqlfluff]
sql_file_exts = .sql
dialect = tsql
templater = placeholder
large_file_skip_byte_limit = 50000
max_line_length = 250

[sqlfluff:templater:placeholder]
param_regex = ${(?P<param_name>[\w_]+)}
V_LOAD_ID = '1111'
V_LOAD_DATE_END = "2020-01-29 08:52:11.0698120"
V_LOAD_DATE_START = "2020-01-29 08:52:11.0698120"
V_TARGET_LAND_SCHEMA = dl
V_TARGET_LAND_TABLE = tbl1
V_LOAD_TYPE = delta
V_REF_LOAD_ID_MIN = '1000'
V_REF_LOAD_ID_MAX = '2000'
V_TARGET_SCHEMA = schema1
V_TARGET_TABLE = table1
PART_SIZE = '10'
SEQUENCE_NUM = '1'

these data below are used for jmx files where it's as default parameters in sql-code
timeStamp = "2020-01-29 08:52:11.0698120"
threadName = threadName
label = label
success = success
failureMessage = failureMessage
responseMessage = responseMessage
Latency = Latency
ldwh_env = LDWH_ETL
test_type = test_type
test_name = test_name
yyyyMMdd = 'yyyyMMdd'
table_name_item = tablename

[sqlfluff:indentation]
indent_unit = tab
tab_space_size = 4
allow_implicit_indents = True

[sqlfluff:rules]
single_table_references = qualified
allow_scalar = False

[sqlfluff:layout:type:bracketed_index_column_list_grammar]
spacing_before = touch

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:layout:type:common_table_expression]
spacing_within = single

[sqlfluff:layout:type:function_name]
spacing_within = touch:inline
spacing_after = touch

[sqlfluff:rules:aliasing.length]
min_alias_length = 2

[sqlfluff:rules:capitalisation.keywords] # Keywords (L010)
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers] # Unquoted identifiers (L014)
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions] # Function names (L030)
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals] # Null & Boolean Literals (L040)
capitalisation_policy = lower

[sqlfluff:rules:references.consistent] # (L028)
single_table_references = qualified

[sqlfluff:rules:capitalisation.types] # datatypes (L063)
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ddbtrmatic** [comment]
Parsing error is present in version `3.2.2` as well.

```sql
DECLARE @result XML;
SET @result = (SELECT
       'Value' as SomeValue
   FOR XML RAW, ROOT('Data'))
```
